### PR TITLE
Bayo izanagi hchukwu

### PR DESCRIPTION
**Pull request by hechukwu**
#### What does this PR do?
* Delivers outputs for:
   * html-js-css-(template)
   * automating tasks
   * writing test

#### Description of Task to be completed?
*a folder containing the template for the proposed frontend of your project* Using Mocha/Jasmine with.
* a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files.
* a pull request showing all test required for the project
#### How should this be manually tested?
* Setting up continuous integration
  * it can be tested at https://github.com/andela-hchukwu/Inverted-Index
```

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)
N/A
#### Questions:PR Naming
N/A
#<story-id> story description
N/A

### CODE REVIEW

File: 06-building-a-program-start-to-finish/writing-tests/output/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Mocha/Jasmine with any assertion library of choice (Chai.JS, Should.JS, Expect.JS...)
+- **Deliver**: a Pull Request showing all tests required for your project
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Test](https://github.com/andela-hchukwu/Inverted-Index/pull/13)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/README.md]: You haven't addressed the comments in this PR.

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+- **With**: *independently*
+
+#### Output
+
+- [Link to folder containing template for proposed frontend of project](https://github.com/andela-hchukwu/Inverted-Index/tree/feature/139456431/front-end-development/src)
+
+#### Screenshots
+
+**_Screenshot of proposed frontend template_**
+![frontend](https://cloud.githubusercontent.com/assets/25608473/23341179/ed3de0e0-fc42-11e6-89ff-8b89f96d283d.png)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md]: Your UI does not show the table that'll display the results.

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Link to YouTube video
+- [Browser reload with Gulp](https://youtu.be/QW1bR0LtsdM)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/automating-tasks/output/README.md:7]: Moving forward, endeavour to mute your mic when recording if you won't be saying anything. There's a lot of background noise.

Other than that, good work!

**Review by q-ode — COMMENTED** [review]
Please address the comments and put test file in your output folder when making the next PR.

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md (hunk 2 above)
Comment (q-ode) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/README.md]: Much better!

**Comment by hechukwu** [comment]
@q-ode is there any reason why this PR was not merged?

File: 06-building-a-program-start-to-finish/writing-tests/output/README.md (hunk 1 above)
Comment (hechukwu) [reply 06-building-a-program-start-to-finish/writing-tests/output/README.md]: I have addressed them

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
+const InvertedIndex = require('../src/inverted-index.js');
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:1]: Don't forget to let your class name `InvertedIndex` match the file name `inverted-index.js`

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      expect(InvertedIndex.tokenize(sampleSentence)).toEqual(['as',
+        'can', 'defined', 'function', 'have', 'here', 'see', 'the',
+        'you', 'you']);
+    });
+    it('should return an array containing the correct number of words', () => {
+      expect(InvertedIndex.tokenize(sampleSentence).length).toBe(10);
+    });
+  });
+
+  describe('Unique Words', () => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:40]: This says unique words, does that mean it does the same as Tokenize?

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+  describe('Populate Index', () => {
+    it('should have an Index created', () => {
+      expect(newIndex.index.books).toBeDefined();
+    });
+    it('should accurately map words to their document location', () => {
+      expect(newIndex.index.books.heroku).toEqual([0]);
+      expect(newIndex.index.books.your).toEqual([0, 1]);
+    });
+    it('should return an object of all created index', () => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:62]: This spec confuses me.

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      const bookKeys = Object.keys(bookIndex);
+      bookKeys.forEach((key) => {
+        expect({}.hasOwnProperty.call(bookIndex, key)).toBeTruthy();
+      });
+    });
+
+    describe('Get Index', () => {
+      it('should return an accurate index Object of the indexed file', () => {
+        expect(newIndex.getIndex('books')).toBeDefined();
+        expect(Object.keys(newIndex.getIndex('books')).length).toBe(37);
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:73]: I'm not sure about the necessity of `Object.keys`.

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        expect(newIndex.searchIndex).toBeDefined();
+      });
+      it('should return no query when no value is passed in', () => {
+        expect(newIndex.searchIndex()).toEqual('no query to search');
+      });
+
+      it('should return an empty {object} if no query is found', () => {
+        expect(typeof newIndex
+      .searchIndex('Checkpoint', 'books')).toEqual('object');
+        expect(Object.keys(newIndex
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:92]: What are you trying to do with `Object.keys`?

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        expect(newIndex.getIndex('books')).toBeDefined();
+        expect(Object.keys(newIndex.getIndex('books')).length).toBe(37);
+      });
+      it('should return undefined if no index has been created', () => {
+        const getIndex = newIndex.getIndex('book3');
+        expect(getIndex).toBeUndefined();
+      });
+    });
+
+    describe('Search index', () => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:81]: In all of this suite I don't see you actually testing for the search results.

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        'can', 'defined', 'function', 'have', 'here', 'see', 'the',
+        'you']);
+    });
+  });
+
+  describe('Populate Index', () => {
+    it('should have an Index created', () => {
+      expect(newIndex.index.books).toBeDefined();
+    });
+    it('should accurately map words to their document location', () => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/output/test.js:58]: It's not adequate that you tested for two words in an entire index. Please test for the exact expected result.

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js (hunk 5 above)
Comment (hechukwu) [reply 06-building-a-program-start-to-finish/writing-tests/output/test.js:40]: yeah it only removes duplicate words, tokenize does not

File: 06-building-a-program-start-to-finish/writing-tests/output/test.js (hunk 8 above)
Comment (hechukwu) [reply 06-building-a-program-start-to-finish/writing-tests/output/test.js:92]: trying to get the length of the object

**Pull request closed without merging**